terraform-modules/aws/opensearch/main.tf

```diff
@@ -0,0 +1,139 @@
+resource "aws_cloudwatch_log_group" "opensearch_slow_logs" {
+  name              = "opensearch-slow-logs"
+  retention_in_days = 14
+}
+
+resource "aws_cloudwatch_log_resource_policy" "opensearch_slow_logs_policy" {
+  policy_name = "opensearch-slow-logs-policy"
+  policy_document = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Principal" : {
+          "Service" : "es.amazonaws.com"
+        },
+        "Action" : [
+          "logs:PutLogEvents",
+          "logs:CreateLogStream",
+          "logs:PutLogEventsBatch"
+        ],
+        "Resource" : "arn:aws:logs:${var.aws_region}:${var.account_id}:log-group:${aws_cloudwatch_log_group.opensearch_slow_logs.name}:*"
+      }
+    ]
+  })
+}
+
+resource "aws_opensearch_domain" "this" {
+  domain_name    = var.domain_name
+  engine_version = "OpenSearch_2.5"
+
+  cluster_config {
+    instance_type          = "r4.large.search"
+    zone_awareness_enabled = true
+    instance_count         = var.instance_count
+  }
+
+  ebs_options {
+    ebs_enabled = true
+    volume_size = 10
+  }
+
+  encrypt_at_rest {
+    enabled = true
+  }
+
+  node_to_node_encryption {
+    enabled = true
+  }
+
+  domain_endpoint_options {
+    enforce_https       = var.enforce_https
+    tls_security_policy = "Policy-Min-TLS-1-2-2019-07"
```

Comment:
Please make this `Policy-Min-TLS-1-2-2019-07` and input var and the default value.  This will change over time and might want to update this without having to change the module.